corehq/apps/style/static/style/ko/components/inline_edit.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Component for an inline editing widget: a piece of text that, when clicked on, turns into a textarea.
+ *
+ * Parameters (all optional)
+ *  - value: Text to display and edit
+ *  - name: HTML name of textarea
+ *  - id: HTML id of textarea
+ *  - placeholder: Text to display when in read-only mode if value is blank
+ *  - inline: Whether or not to display widget in line with surrounding content. Defaults to false.
+ *  - rows: Number of rows in textarea.
+ *
+ * By default, the widget is client-side only, and it is up to the calling code to actually save the value
```

Comment:
Guessing this means that by default the user has to click the green checkbox after editing and then also submit the form elsewhere on the page. This seems like a not great default user experience—I think most users would expect the value to be saved after clicking the green checkbox. I'm not sure I have any good suggestions to work around that issue, so this is mostly just food for thought.